README
NAME
    Text::Pipe::Encoding - Text pipes that can encode and decode in various
    ways

VERSION
    version 1.100950

DESCRIPTION
    This is a marker class; the actual pipe segment classes live in the
    "Text::Pipe::Encoding::" namespace.

INSTALLATION
    See perlmodinstall for information and options on installing Perl
    modules.

BUGS AND LIMITATIONS
    No bugs have been reported.

    Please report any bugs or feature requests through the web interface at
    <http://rt.cpan.org/Public/Dist/Display.html?Name=Text-Pipe-Encoding>.

AVAILABILITY
    The latest version of this module is available from the Comprehensive
    Perl Archive Network (CPAN). Visit <http://www.perl.com/CPAN/> to find a
    CPAN site near you, or see
    <http://search.cpan.org/dist/Text-Pipe-Encoding/>.

    The development version lives at
    <http://github.com/hanekomu/Text-Pipe-Encoding/>. Instead of sending
    patches, please fork this project using the standard git and github
    infrastructure.

AUTHOR
      Marcel Gruenauer <marcel@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2007 by Marcel Gruenauer.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Revision history for Perl extension Text-Pipe-Encoding

1.100950  2010-04-05 00:21:30 Europe/Vienna
    - require MIME-Base64 3.09 since I adapted the tests from there

1.100880  2010-03-29 23:32:52 Europe/Vienna
    - converted the distribution to Dist::Zilla-style

0.03  Thu Jul 25 15:27:43 CEST 2008 (Marcel Gruenauer <marcel@cpan.org>)
    - Changed dist style and Changes back to standard. People didn't like it -
      the nail that sticks out gets hammered down.
    - Added standard test files; this will also help with CPANTS.

0.02  Thu, 29 May 2008 12:08:48 -0000 (Marcel Gruenauer <marcel@cpan.org>)
    - set the version to 0.02
    - .shipit: added Twitter step
    - Makefile.PL: added auto_install()
    - updated MANIFEST
    - t/02_quoted_print.t: removed commented-out code
    - updated t/perlcriticrc
    - moved generated documentation of interited methods to after documentation
      of this class' methods so the manpage is more readable
    - .shipit: fixed svk.tagpattern

0.01  Sun, 11 Nov 2007 22:18:52 -0000 (Marcel Gruenauer <marcel@cpan.org>)
    - original version


lib/Text/Pipe/Encoding.pm
use 5.008;
use strict;
use warnings;

package Text::Pipe::Encoding;
BEGIN {
  $Text::Pipe::Encoding::VERSION = '1.100950';
}
# ABSTRACT: Text pipes that can encode and decode in various ways
use parent qw(Text::Pipe::Base);
1;


__END__
=pod

=head1 NAME

Text::Pipe::Encoding - Text pipes that can encode and decode in various ways

=head1 VERSION

version 1.100950

=head1 DESCRIPTION

This is a marker class; the actual pipe segment classes live in the
C<Text::Pipe::Encoding::> namespace.

=head1 INSTALLATION

See perlmodinstall for information and options on installing Perl modules.

=head1 BUGS AND LIMITATIONS

No bugs have been reported.

Please report any bugs or feature requests through the web interface at
L<http://rt.cpan.org/Public/Dist/Display.html?Name=Text-Pipe-Encoding>.

=head1 AVAILABILITY

The latest version of this module is available from the Comprehensive Perl
Archive Network (CPAN). Visit L<http://www.perl.com/CPAN/> to find a CPAN
site near you, or see
L<http://search.cpan.org/dist/Text-Pipe-Encoding/>.

The development version lives at
L<http://github.com/hanekomu/Text-Pipe-Encoding/>.
Instead of sending patches, please fork this project using the standard git
and github infrastructure.

=head1 AUTHOR

  Marcel Gruenauer <marcel@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2007 by Marcel Gruenauer.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut



inc/Module/Install/Template.pm
#line 1
package Module::Install::Template;

use strict;
use warnings;
use File::Temp 'tempfile';
use YAML;


our $VERSION = '0.03';


use base qw(Module::Install::Base);


sub is_author {
    my $author = $^O eq 'VMS' ? './inc/_author' : './inc/.author';
    -d $author;
}


sub tag {
    my $self = shift;
    (my $name = lc $self->name) =~ s/-//g;
    $name;
}


sub rt_email {
    my $self = shift;
    sprintf '<bug-%s@rt.cpan.org>', lc $self->name;
}


sub year_str {
    my ($self, $first_year) = @_;
    my $this_year = ((localtime)[5] + 1900);
    return $this_year if (!defined $first_year) || $first_year == $this_year;
    die "first year ($first_year) is after this year ($this_year)?\n"
        if $first_year > $this_year;
    return "$first_year-$this_year";
}


sub process_templates {
    my ($self, %args) = @_;

    $self->build_requires('YAML');

    # only module authors should process templates; if you're not the original
    # author, you won't have the templates anyway, only the generated files.

    return unless $self->is_author;

    $::WANTS_MODULE_INSTALL_TEMPLATE = 1;

    my $config = {
        template => {
            INCLUDE_PATH => "$ENV{HOME}/.mitlib",
            (defined $args{start_tag} ? (START_TAG => $args{start_tag}) : ()),
            (defined $args{end_tag}   ? (END_TAG   => $args{end_tag})   : ()),
        },
        vars => {
            name     => $self->name,
            year     => $self->year_str($args{first_year}),
            tag      => $self->tag,
            rt_email => $self->rt_email,
        },
    };

    my ($fh, $filename) = tempfile();
    print $fh Dump $config;
    close $fh or die "can't close $filename: $!\n";

    $self->makemaker_args(PM_FILTER => "tt_pm_to_blib $filename");

    # Some of the following may not have been available in the template; the
    # module author can specify that they should come from somewhere else.

    if (defined $args{rest_from}) {

        # try to get all values that haven't been defined yet from the
        # indicated source

        for my $key (qw(version perl_version author license abstract)) {
            next if defined($self->$key) && length($self->$key);
            my $method = "${key}_from";
            $self->$method($args{rest_from});
        }
    }

    $self->all_from($args{all_from})
        if defined $args{all_from};

    $self->version_from($args{version_from})
        if defined $args{version_from};

    $self->perl_version_from($args{perl_version_from})
        if defined $args{perl_version_from};

    $self->author_from($args{author_from})
        if defined $args{author_from};

    $self->license_from($args{license_from})
        if defined $args{license_from};

    $self->abstract_from($args{abstract_from})
        if defined $args{abstract_from};
}


# 'make dist' uses ExtUtils::Manifest's maniread() and manicopy() to determine
# what should be copied into the dist dir. This is fine for most purposes, but
# with Moduile::Install::Template we really want the finished files, not the
# templates. So we override the create_distdir rule here, but only if we're
# the author - the end user shouldn't be bothered with any of this
# kludge^Wmagic.
#
# We still use maniread() to get the file names, but then change those
# filenames pointing into lib/ to point into blib/lib instead. Now the right
# files get copied, but they end up in the wrong place - in the dist dir's
# blib/lib/. So at the end we move them to the right place and delete the dist
# dir's blib/ directory.
#
# And 'make disttest' needs to be modified as well; we need to have the blib/
# files before we can test the distribution. So I've added a 'pm_to_blib'
# requirement to the 'disttest' target.


sub MY::postamble {
    my $self = shift;
    return '' if defined $::IS_MODULE_INSTALL_TEMPLATE;

    # for some reason, Module::Install runs this subroutine even if the
    # Makefile.PL doesn't specify process_template(). So here we check whether
    # process_template() has been run.

    return '' unless defined $::WANTS_MODULE_INSTALL_TEMPLATE;

    return '' unless Module::Install::Template->is_author;
    return <<'EOPOSTAMBLE';
create_distdir : pm_to_blib
	$(RM_RF) $(DISTVNAME)
	$(PERLRUN) "-MExtUtils::Manifest=manicopy,maniread" \
	    -e '$$m = maniread(); while (($$k, $$v) = each %$$m) { next if $$k !~ m!^lib/!; delete $$m->{$$k}; $$m->{"blib/$$k"} = $$v; }; manicopy($$m, "$(DISTVNAME)", "$(DIST_CP)"); '
	$(MV) $(DISTVNAME)/blib/lib $(DISTVNAME)/lib
	$(RM_RF) $(DISTVNAME)/blib

disttest : pm_to_blib distdir
	cd $(DISTVNAME) && $(ABSPERLRUN) Makefile.PL 
	cd $(DISTVNAME) && $(MAKE) $(PASTHRU)
	cd $(DISTVNAME) && $(MAKE) test $(PASTHRU)
EOPOSTAMBLE
}


1;


__END__

#line 251



lib/Text/Pipe/Encoding/QuotedPrint/Decode.pm
use 5.008;
use strict;
use warnings;

package Text::Pipe::Encoding::QuotedPrint::Decode;
BEGIN {
  $Text::Pipe::Encoding::QuotedPrint::Decode::VERSION = '1.100950';
}
# ABSTRACT: Text pipe that can decode quoted-printable strings
use MIME::QuotedPrint 3.09;
use parent qw(Text::Pipe::Encoding);

sub filter_single {
    my ($self, $input) = @_;
    decode_qp($input);
}
1;


__END__
=pod

=head1 NAME

Text::Pipe::Encoding::QuotedPrint::Decode - Text pipe that can decode quoted-printable strings

=head1 VERSION

version 1.100950

=head1 SYNOPSIS

    my $pipe = Text::Pipe->new('Encoding::Base64::Decode');
    my $result = $pipe->filter(
        # ...
    );

=head1 METHODS

=head2 filter_single

Takes a single string, decodes it as quoted-printable, and returns it. See
L<MIME::QuotedPrint> for details.

=head1 INSTALLATION

See perlmodinstall for information and options on installing Perl modules.

=head1 BUGS AND LIMITATIONS

No bugs have been reported.

Please report any bugs or feature requests through the web interface at
L<http://rt.cpan.org/Public/Dist/Display.html?Name=Text-Pipe-Encoding>.

=head1 AVAILABILITY

The latest version of this module is available from the Comprehensive Perl
Archive Network (CPAN). Visit L<http://www.perl.com/CPAN/> to find a CPAN
site near you, or see
L<http://search.cpan.org/dist/Text-Pipe-Encoding/>.

The development version lives at
L<http://github.com/hanekomu/Text-Pipe-Encoding/>.
Instead of sending patches, please fork this project using the standard git
and github infrastructure.

=head1 AUTHOR

  Marcel Gruenauer <marcel@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2007 by Marcel Gruenauer.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut



lib/Text/Pipe/Encoding/QuotedPrint/Encode.pm
use 5.008;
use strict;
use warnings;

package Text::Pipe::Encoding::QuotedPrint::Encode;
BEGIN {
  $Text::Pipe::Encoding::QuotedPrint::Encode::VERSION = '1.100950';
}
# ABSTRACT: Text pipe that can encode quoted-printable strings
use MIME::QuotedPrint 3.09;
use parent qw(Text::Pipe::Encoding);

sub filter_single {
    my ($self, $input) = @_;
    encode_qp($input);
}
1;


__END__
=pod

=head1 NAME

Text::Pipe::Encoding::QuotedPrint::Encode - Text pipe that can encode quoted-printable strings

=head1 VERSION

version 1.100950

=head1 SYNOPSIS

    my $pipe = Text::Pipe->new('Encoding::Base64::Encode');
    my $result = $pipe->filter(
        # ...
    );

=head1 METHODS

=head2 filter_single

Takes a single string, encodes it as quoted-printable, and returns it. See
L<MIME::QuotedPrint> for details.

=head1 INSTALLATION

See perlmodinstall for information and options on installing Perl modules.

=head1 BUGS AND LIMITATIONS

No bugs have been reported.

Please report any bugs or feature requests through the web interface at
L<http://rt.cpan.org/Public/Dist/Display.html?Name=Text-Pipe-Encoding>.

=head1 AVAILABILITY

The latest version of this module is available from the Comprehensive Perl
Archive Network (CPAN). Visit L<http://www.perl.com/CPAN/> to find a CPAN
site near you, or see
L<http://search.cpan.org/dist/Text-Pipe-Encoding/>.

The development version lives at
L<http://github.com/hanekomu/Text-Pipe-Encoding/>.
Instead of sending patches, please fork this project using the standard git
and github infrastructure.

=head1 AUTHOR

  Marcel Gruenauer <marcel@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2007 by Marcel Gruenauer.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut



Makefile.PL

use strict;
use warnings;

 BEGIN { require 5.008; } 

use ExtUtils::MakeMaker 6.31;



my %WriteMakefileArgs = (
                       'test' => {
                                   'TESTS' => 't/*.t'
                                 },
                       'NAME' => 'Text::Pipe::Encoding',
                       'DISTNAME' => 'Text-Pipe-Encoding',
                       'CONFIGURE_REQUIRES' => {
                                                 'ExtUtils::MakeMaker' => '6.31'
                                               },
                       'AUTHOR' => 'Marcel Gruenauer <marcel@cpan.org>',
                       'BUILD_REQUIRES' => {
                                             'Scalar::Util' => '0',
                                             'English' => '0',
                                             'Encode' => '0',
                                             'Carp' => '0',
                                             'Test::More' => '0.88',
                                             'File::Find' => '0',
                                             'File::Temp' => '0',
                                             'Text::Pipe' => '0'
                                           },
                       'ABSTRACT' => 'Text pipes that can encode and decode in various ways',
                       'EXE_FILES' => [],
                       'VERSION' => '1.100950',
                       'PREREQ_PM' => {
                                        'parent' => '0',
                                        'MIME::Base64' => '3.09',
                                        'Text::Pipe::Base' => '0',
                                        'MIME::QuotedPrint' => '3.09'
                                      },
                       'LICENSE' => 'perl'
                     );


delete $WriteMakefileArgs{BUILD_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.56) };

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



